### PR TITLE
feat: negation of `Intervals`

### LINKED ISSUE #6311: Negation of `Intervals`
### Is your feature request related to a problem or challenge?

I propose to introduce the negation of `Intervals` into `arrow-datafusion`.
Example:
```
query ?
select -interval '5' year;
----
0 years -60 mons 0 days 0 hours 0 mins 0.000000000 secs
```

### Describe the solution you'd like

The function `negative` and `arithmetic_negate` starts to support all types of `Intervals`.

### Describe alternatives you've considered

_No response_

### Additional context

PostgreSQL (Date/Time Functions and Operators): https://www.postgresql.org/docs/current/functions-datetime.html

### PR DESCRIPTION
**Pull request by izveigor**
# Which issue does this PR close?

Closes #6311

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?
Yes

# Are there any user-facing changes?
Yes

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/interval.slt [hunk 1]
```diff
@@ -186,6 +186,30 @@ select interval '1 year' - '1 month' - '1 day' - '1 hour' - '1 minute' - '1 seco
 ----
 0 years 11 mons -1 days -1 hours -1 mins -1.001001001 secs
 
+# Interval with string literal negation and leading field
+query ?
+select -interval '5' - '1' - '2' year;
+----
+0 years -24 mons 0 days 0 hours 0 mins 0.000000000 secs
```
[the remark this answers is not in the document]
Comment (comphead) [reply datafusion/core/tests/sqllogictests/test_files/interval.slt:193]:
```
select extract( year from interval '5' year);
```
This works in PG and returns 5 years 

But fails in DF, I'll create another ticket on this.

[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/interval.slt:193]: I think it is related to the fact that interval gets parsed to the same IntervalMonthDayNano all the time -- which was changed in https://github.com/apache/arrow-datafusion/pull/5806

I think to handle it properly we need to have cast support between the different interval types, maybe 🤔

File: datafusion/physical-expr/src/expressions/negative.rs [hunk 2]
```diff
@@ -145,11 +151,88 @@ pub fn negative(
...
+    use crate::expressions::col;
+    #[allow(unused_imports)]
+    use arrow::array::*;
+    use arrow::datatypes::*;
+    use arrow_schema::DataType::{Float32, Float64, Int16, Int32, Int64, Int8};
+    use arrow_schema::IntervalUnit::{DayTime, MonthDayNano, YearMonth};
+    use datafusion_common::{cast::as_primitive_array, Result};
+    use paste::paste;
+
+    macro_rules! test_array_negative_op {
```
Comment (alamb) [inline datafusion/physical-expr/src/expressions/negative.rs:175]: thank you for adding this coverage of regular negate

**Review by alamb — APPROVED** [review]
Looks like an improvement to me -- thank you @izveigor  and @comphead

**Comment by alamb** [comment]
Looks like @comphead has filed https://github.com/apache/arrow-datafusion/issues/6327 👍

**Pull request merged**